### PR TITLE
[SPARK-9100] [SQL] Adds DataFrame reader/writer shortcut methods for ORC

### PR DESCRIPTION
**Pull request by liancheng**
This PR adds DataFrame reader/writer shortcut methods for ORC in both Scala and Python.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala [hunk 1]
```diff
@@ -268,6 +268,14 @@ class DataFrameReader private[sql](sqlContext: SQLContext) {
   }
 
   /**
+   * Loads an ORC file and returns the result as a [[DataFrame]].
+   *
```
Comment (chenghao-intel) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala]: Nit: Also add examples like you did at DataFrameWriter?

File: python/pyspark/sql/readwriter.py [hunk 2]
```diff
@@ -146,14 +146,24 @@ def table(self, tableName):
...
 
         >>> df = sqlContext.read.parquet('python/test_support/sql/parquet_partitioned')
         >>> df.dtypes
         [('name', 'string'), ('year', 'int'), ('month', 'int'), ('day', 'int')]
         """
-        return self._df(self._jreader.parquet(_to_seq(self._sqlContext._sc, path)))
+        return self._df(self._jreader.parquet(_to_seq(self._sqlContext._sc, paths)))
+
+    @since(1.5)
+    def orc(self, path):
```
Comment (chenghao-intel) [inline python/pyspark/sql/readwriter.py]: As the function `parquet` support multiple path in loading, should the api `orc` support that also?

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala [hunk 3]
```diff
@@ -268,6 +268,14 @@ class DataFrameReader private[sql](sqlContext: SQLContext) {
...
+   * @param path input path
+   * @since 1.5.0
+   */
+  def orc(path: String): DataFrame = format("orc").load(path)
```
Comment (chenghao-intel) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala]: Multiple paths support?

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala (hunk 1 above)
Comment (liancheng) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala]: The one in `DataFrameWriter` isn't an example, it says that method is equivalent to `format("orc").load(path)`.

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala (hunk 3 above)
Comment (liancheng) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala]: I did consider this but didn't add multiple-path support in this PR intentionally. The problem here is that currently we can't specify multiple paths via data source options, while DataFrame reader/writer API relies on the "path" option to find the input path. If you check `DataFrameReader.parquet`, you may see it's specially implemented, which I don't think is a good approach.

I'm thinking about adding multiple value support for data source options in a more general way. (Maybe just simple comma separated lists with proper comma escaping.)

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala [hunk 4]
```diff
@@ -280,6 +280,17 @@ final class DataFrameWriter private[sql](df: DataFrame) {
    */
   def parquet(path: String): Unit = format("parquet").save(path)
 
+  /**
+   * Saves the content of the [[DataFrame]] in ORC format at the specified path.
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala]: you should document this is only available if hive is ...

Comment (liancheng) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala]: Yeah, right. Thanks for reminding!

**Comment by liancheng** [comment]
@rxin @davies A little off topic, but `python/pyspark/sql/readwriter.py` should probably be renamed to `readerwriter.py` or `readwrite.py`.  Which do you prefer?

**Comment by liancheng** [comment]
retest please

**Comment by liancheng** [comment]
retest this please

**Comment by rxin** [comment]
lgtm

**Comment by liancheng** [comment]
Thanks for the review, merging to master.

**Pull request closed without merging**